### PR TITLE
Support Remote File Browser

### PR DESCRIPTION
**Pull request by perspectivus1**
+ other custom question types (inquirer-gui plugins)

### CODE REVIEW

File: backend/src/yeomanui.ts [hunk 1]
```diff
@@ -377,9 +400,22 @@ export class YeomanUI {
...
    */
   private normalizeFunctions(questions: Environment.Adapter.Questions<any>): Environment.Adapter.Questions<any> {
+    this.addCustomQuestionEventHandlers(questions);
     return JSON.parse(JSON.stringify(questions, YeomanUI.funcReplacer));
   }
 
+  private addCustomQuestionEventHandlers(questions: Environment.Adapter.Questions<any>): void {
+    for (const index in questions) {
+      const question = (questions as any[])[Number.parseInt(index)];
+      const questionHandlers = this.customQuestionEventHandlers.get((question as any)["guiType"]);
```
Comment (tomer-epstein) [inline backend/src/yeomanui.ts]: isn't question can have only 1 guiType?

Comment (perspectivus1) [reply backend/src/yeomanui.ts]: Yes, only 1 guiType, but here we're looping through all questions of a particular prompt.

Then, each question can possibly have several event handlers. For example, login and logout for a custom login question type.

File: backend/src/extension.ts [hunk 2]
```diff
@@ -98,6 +98,25 @@ export class YeomanUIPanel {
...
+			uri = vscode.Uri.file(currentPath);
+		} catch (e) {
+			uri = vscode.Uri.file('/');
+		}
+
+		try {
+			const filePath = await vscode.window.showOpenDialog({
+				canSelectFiles: true,
+				defaultUri: uri
+			});
```
Comment (tomer-epstein) [inline backend/src/extension.ts]: In Theia 'prompt 2' of generator-foodq is empty.

image

File: frontend/package.json [hunk 3]
```diff
@@ -18,7 +18,7 @@
     "@mdi/font": "^4.9.95",
     "@sap-devx/inquirer-gui": "0.0.31",
     "@sap-devx/inquirer-gui-login-plugin": "0.0.1",
-    "@sap-devx/inquirer-gui-remote-file-browser-plugin": "0.0.1",
+    "@sap-devx/inquirer-gui-remote-file-browser-plugin": "0.0.2",
```
Comment (tomer-epstein) [inline frontend/package.json]: Why do I have to specific declare dependency to inquirer-gui-remote-file-browser-plugin / inquirer-gui-login-plugin ?
I would expect all plugin to be available within -> "@sap-devx/inquirer-gui"

Comment (perspectivus1) [reply frontend/package.json]: It's a matter of definition. Currently only controls for [built-in inquirer prompt types](https://github.com/SBoudrias/Inquirer.js#prompt-types) are provided by default in inquirer-gui.  

We could change that behavior and provide custom controls (inquirer-gui plugins) that we develop together with the package, but that's a matter that product should decide.

Comment (tomer-epstein) [reply frontend/package.json]: OK. we'll considure this in future.

**Pull request merged**